src/content/app/genome-browser/state/browser-general/browserGeneralSlice.ts

```diff
@@ -0,0 +1,301 @@
+    browserStorageService.updateActiveEnsObjectIds({
+      [activeGenomeId]: activeEnsObjectId
+    });
+  }
+};
+
+export const updateBrowserActiveEnsObjectIdsAndSave = (
+  activeEnsObjectId: string
+): ThunkAction<void, any, null, Action<string>> => {
+  return (dispatch, getState: () => RootState) => {
+    const state = getState();
+    const activeGenomeId = getBrowserActiveGenomeId(state);
+    if (!activeGenomeId) {
+      return;
+    }
+    const currentActiveEnsObjectIds = getBrowserActiveEnsObjectIds(state);
+    const updatedActiveEnsObjectIds = {
+      ...currentActiveEnsObjectIds,
+      [activeGenomeId]: activeEnsObjectId
+    };
+
+    dispatch(updateBrowserActiveEnsObjectIds(updatedActiveEnsObjectIds));
+    dispatch(fetchEnsObject(activeEnsObjectId));
+
+    browserStorageService.updateActiveEnsObjectIds(updatedActiveEnsObjectIds);
+  };
+};
+
+export const setChrLocation: ActionCreator<
+  ThunkAction<any, any, null, Action<string>>
+> = (chrLocation: ChrLocation) => {
+  return (dispatch, getState: () => RootState) => {
+    const state = getState();
+    const activeGenomeId = getBrowserActiveGenomeId(state);
+    const activeEnsObjectId = getBrowserActiveEnsObjectId(state);
+
+    if (!activeGenomeId || !activeEnsObjectId) {
+      return;
+    }
+    const savedChrLocation = getChrLocation(state);
+
+    const payload = {
+      [activeGenomeId]: chrLocation
+    };
+
+    dispatch(updateChrLocation(payload));
+    browserStorageService.updateChrLocation(payload);
+
+    if (!isEqual(chrLocation, savedChrLocation)) {
+      const newUrl = urlFor.browser({
+        genomeId: activeGenomeId,
+        focus: buildFocusIdForUrl(activeEnsObjectId),
+        location: getChrLocationStr(chrLocation)
+      });
+      dispatch(replace(newUrl));
+    }
+  };
+};
+
+export const deleteSpeciesInGenomeBrowser = (
+  genomeIdToRemove: string
+): ThunkAction<void, any, null, Action<string>> => {
+  return (dispatch, getState: () => RootState) => {
+    const state = getState();
+
+    dispatch(deleteGenomeGeneralData(genomeIdToRemove));
+    dispatch(deleteGenomeTrackPanelData(genomeIdToRemove));
+
+    const updatedActiveEnsObjectIds = pickBy(
+      getBrowserActiveEnsObjectIds(state),
+      (value, key) => key !== genomeIdToRemove
+    );
+
+    dispatch(updateBrowserActiveEnsObjectIds(updatedActiveEnsObjectIds));
+
+    browserStorageService.deleteGenome(genomeIdToRemove);
+    trackPanelStorageService.deleteGenome(genomeIdToRemove);
+  };
+};
+
+export const defaultBrowserGeneralState: BrowserGeneralState = {
+  activeGenomeId: null,
+  activeEnsObjectIds: {},
+  trackStates: {},
+  chrLocations: {},
+  actualChrLocations: {},
+  regionEditorActive: false,
+  regionFieldActive: false,
+  isObjectInDefaultPosition: false
+};
+
+const browserGeneralSlice = createSlice({
+  name: 'genome-browser-general',
+  initialState: defaultBrowserGeneralState as BrowserGeneralState,
+  reducers: {
+    loadBrowserGeneralState(state) {
```

Comment:
you didn't put it in a thunk after all?

Reply:
Sorry, missed some bits there. That's done now.